### PR TITLE
Resolves issue #34

### LINKED ISSUE #34: Image uploading API with file should be multipart/form-data header
As title, when using the Imgur client to upload image, it should throw following exception:

```
PHP Fatal error:  Uncaught Imgur\Exception\ErrorException: Request to: /3/image failed with: "Invalid URL (@Resource id #50)" in /data/php-imgur-api-client/lib/Imgur/Middleware/ErrorMiddleware.php:91
Stack trace:
#0 /data/php-imgur-api-client/lib/Imgur/Middleware/ErrorMiddleware.php(35): Imgur\Middleware\ErrorMiddleware->checkError(Object(GuzzleHttp\Psr7\Response))
#1 /data/php-imgur-api-client/vendor/guzzlehttp/promises/src/FulfilledPromise.php(41): Imgur\Middleware\ErrorMiddleware->Imgur\Middleware\{closure}(Object(GuzzleHttp\Psr7\Response))
#2 /data/php-imgur-api-client/vendor/guzzlehttp/promises/src/TaskQueue.php(48): GuzzleHttp\Promise\FulfilledPromise::GuzzleHttp\Promise\{closure}()
#3 /data/php-imgur-api-client/vendor/guzzlehttp/promises/src/Promise.php(248): GuzzleHttp\Promise\TaskQueue->run(true)
#4 /data/php-imgur-api-client/vendor/guzzlehttp/promises/src/Promise.php(224): GuzzleHttp\Promise\Promise->invokeWaitFn()
#5 /data/php-imgur-api-client/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promi in /data/php-imgur-api-client/lib/Imgur/Middleware/ErrorMiddleware.php on line 91
```

The reproduced codes are as follows:

```PHP

<?php

require __DIR__ . '/vendor/autoload.php';

use Imgur\Client;

$client = new Client();

$client->setOption('client_id', 'client_id');
$client->setOption('client_secret', 'client_secret');

$test = $client->api('image')->upload([
    'image' => __DIR__ . '/1.jpg',
    'type' => 'file',
]);
var_dump($test);

```

After looking at the [official Imgur API doc](https://apidocs.imgur.com/#c85c9dfc-7487-4de2-9ecd-66f727cf3139), using the image file path and type is file should use the `multipart/form-data` to be the request header, not the `form_params` (the `application/x-www-form-urlencoded` header).

### PR DESCRIPTION
**Pull request by peter279k**
Closes #34.

### CODE REVIEW

File: lib/Imgur/Api/Image.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ public function upload($data)
         }
 
         if ('file' === $data['type']) {
-            $data['image'] = '@' . $data['image'];
```
Comment (j0k3r) [inline lib/Imgur/Api/Image.php:52]: The `@` tricks is working with curl https://www.iditect.com/how-to/56655862.html
But I don't know with Guzzle.

**Review by j0k3r — COMMENTED** [review]
Could you fix tests?

**Comment by peter279k** [comment]
The failed test is as follows:

```
1) Imgur\tests\Api\ImageTest::testUploadWithFile
fopen(./ZOY11VC.png): failed to open stream: No such file or directory
/home/travis/build/peter279k/php-imgur-api-client/lib/Imgur/Api/Image.php:52
/home/travis/build/peter279k/php-imgur-api-client/tests/Api/ImageTest.php:144
```

Download the `http://i.imgur.com/ZOY11VC.png` to the local image file on `tests/Api/ZOY11VC.png` folder and let this file be under Git version control.

File: .travis.yml [hunk 2]
```diff
@@ -33,7 +33,7 @@ before_script:
     - composer install --prefer-dist --no-interaction -o
 
 script:
-    - ./vendor/bin/simple-phpunit -v --coverage-clover=coverage.clover
+    - XDEBUG_MODE=coverage ./vendor/bin/simple-phpunit -v --coverage-clover=coverage.clover
```
Comment (peter279k) [inline .travis.yml:36]: This fix is issued by following error:

```
PHP Fatal error:  Uncaught SebastianBergmann\CodeCoverage\RuntimeException: XDEBUG_MODE=coverage or xdebug.mode=coverage has to be set in /home/travis/build/peter279k/php-imgur-api-client/vendor/bin/.phpunit/phpunit-8.3-0/vendor/phpunit/php-code-coverage/src/Driver/Xdebug.php:50
Stack trace:
....
```

And the reference is available [here](https://github.com/sebastianbergmann/php-code-coverage/issues/834).

**Comment by peter279k** [comment]
Fix coding style via `PHP-CS-Fixer` during `PHP-7.2` coding style checking:

```
4.16s$ if [ "$CS_FIXER" = "run" ]; then php vendor/bin/php-cs-fixer fix --verbose --dry-run ; fi;
Loaded config default from "/home/travis/build/peter279k/php-imgur-api-client/.php_cs".
Runtime: PHP 7.2.15
F..F.F..........FF.......F.F..F.FFFFFFF.F.F...
Legend: ?-unknown, I-invalid file syntax (file ignored), S-skipped (cached or empty file), .-no changes, F-fixed, E-error
   1) lib/Imgur/Auth/OAuth2.php (no_superfluous_phpdoc_tags, phpdoc_align)
   2) lib/Imgur/Api/Album.php (no_superfluous_phpdoc_tags)
   3) lib/Imgur/Api/CustomGallery.php (no_superfluous_phpdoc_tags, phpdoc_trim_consecutive_blank_line_separation)
   4) lib/Imgur/Middleware/ErrorMiddleware.php (no_superfluous_phpdoc_tags, phpdoc_trim)
   5) lib/Imgur/Middleware/AuthMiddleware.php (no_superfluous_phpdoc_tags, phpdoc_trim)
   6) lib/Imgur/HttpClient/HttpClient.php (no_superfluous_phpdoc_tags)
   7) lib/Imgur/Client.php (no_superfluous_phpdoc_tags, no_empty_phpdoc, no_whitespace_in_blank_line, no_extra_blank_lines)
   8) tests/Api/GalleryTest.php (php_unit_mock_short_will_return)
   9) tests/Api/AlbumTest.php (php_unit_mock_short_will_return)
  10) tests/Api/NotificationTest.php (php_unit_mock_short_will_return)
  11) tests/Api/CustomGalleryTest.php (php_unit_mock_short_will_return)
  12) tests/Api/AccountTest.php (php_unit_mock_short_will_return)
  13) tests/Api/CommentTest.php (php_unit_mock_short_will_return)
  14) tests/Api/ConversationTest.php (php_unit_mock_short_will_return)
  15) tests/Api/ImageTest.php (php_unit_mock_short_will_return)
  16) tests/Api/TopicTest.php (php_unit_mock_short_will_return)
  17) tests/Api/MemegenTest.php (php_unit_mock_short_will_return)

```

**Comment by j0k3r** [comment]
Good job!

**Review by j0k3r — APPROVED** [review]

**Pull request merged**